x/auth/client/cli/sign.go

```diff
@@ -122,35 +125,58 @@ func makeSignCmd(cdc *amino.Codec) func(cmd *cobra.Command, args []string) error
 		if err != nil {
 			return err
 		}
+
 		defer fp.Close()
 		fmt.Fprintf(fp, "%s\n", json)
+
 		return
 	}
 }
 
-func printSignatures(stdTx auth.StdTx) bool {
+func printAndValidateSigs(cliCtx context.CLIContext, chainID string, stdTx auth.StdTx) bool {
```

Comment:
The `--offline` flag in this case wouldn't take any effect anymore. This could potentially be addressed by either:

1. modifyng the `--offline` flags docs stating that `--validate-signatures` requires access to cache, therefore the offline flag would be ignored (x/auth/client/cli/sign.go:55). I'd recommend to drop an explainer in the command's long description too.
2. `--offline` and `--validate-signatures` could even work together, i.e. when both flags are on the code runs just a lightweight signature check:

```
		if i >= len(signers) || !sigAddr.Equals(signers[i]) {
			sigSanity = fmt.Sprintf("ERROR: signature %d does not match its respective signer", i)
```

Reply:
Updated! 🍍